tests/openqa_bats_review_test.py

```diff
@@ -0,0 +1,647 @@
+            headers={
+                "User-Agent": "openqa-bats-review (https://github.com/os-autoinst/scripts)"
+            },
+            timeout=30,
+        )
+        mock_response.raise_for_status.assert_called_once()
+
+    @patch("bats_review.session")
+    @patch("bats_review.log")
+    def test_get_file_request_exception(self, mock_log, mock_session):
+        """Test handling of request exceptions"""
+        mock_session.get.side_effect = RequestException("Network error")
+
+        with pytest.raises(SystemExit) as exc_info:
+            get_file("http://example.com/test.txt")
+
+        assert exc_info.value.code == 1
+        mock_log.error.assert_called_once()
+
+    @patch("bats_review.session")
+    @patch("bats_review.log")
+    def test_get_file_http_error(self, mock_log, mock_session):
+        """Test handling of HTTP errors"""
+        mock_response = Mock()
+        mock_response.raise_for_status.side_effect = RequestException("404 Not Found")
+        mock_session.get.return_value = mock_response
+
+        with pytest.raises(SystemExit) as exc_info:
+            get_file("http://example.com/test.txt")
+
+        assert exc_info.value.code == 1
+        mock_log.error.assert_called_once()
+
+
+class TestGetJob:
+    """Test the get_job function"""
+
+    def setup_method(self):
+        """Clear cache before each test"""
+        get_job.cache_clear()
+
+    @patch("bats_review.session")
+    def test_get_job_success(self, mock_session):
+        """Test successful job retrieval"""
+        mock_response = Mock()
+        mock_response.json.return_value = {"job": {"id": 123, "state": "done"}}
+        mock_session.get.return_value = mock_response
+
+        result = get_job("http://example.com/api/v1/jobs/123")
+
+        assert result == {"id": 123, "state": "done"}
+        mock_session.get.assert_called_once_with(
+            "http://example.com/api/v1/jobs/123",
+            headers={
+                "User-Agent": "openqa-bats-review (https://github.com/os-autoinst/scripts)"
+            },
+            timeout=30,
+        )
+
+    @patch("bats_review.session")
+    @patch("bats_review.log")
+    def test_get_job_request_exception(self, mock_log, mock_session):
+        """Test handling of request exceptions in get_job"""
+        mock_session.get.side_effect = RequestException("Network error")
+
+        with pytest.raises(SystemExit) as exc_info:
+            get_job("http://example.com/api/v1/jobs/123")
+
+        assert exc_info.value.code == 1
+        mock_log.error.assert_called_once()
+
+
+class TestGrepNotok:
+    """Test the grep_notok function"""
+
+    @patch("bats_review.get_file")
+    @patch("bats_review.log")
+    def test_grep_notok_success(self, mock_log, mock_get_file):
+        """Test successful parsing of TAP file"""
+        tap_content = """1..3
+ok 1 test one
+not ok 2 failing test in 100ms
+ok 3 test three"""
+
+        mock_get_file.return_value = tap_content
+
```

Comment:
I'd avoid all these excessive blank lines.

Reply:
> I'd avoid all these excessive blank lines.

If I remove them `ruff` will complain.  I used `black` for formatting.